### PR TITLE
Update README to remove recommendation of `create_with_primary_key` configuration

### PR DESCRIPTION
**Pull request by webandtech**
https://github.com/rkrage/pg_party/issues/57 raised the concern that ActiveRecord does not natively support composite primary keys, yet using the `create_with_primary_key` option will most often require using them (unless `partition_key:` is `:id`).

* Remove recommendation that `create_with_primary_key` option be used with Postgres 11+ and instead document the caveat that composite primary keys are not supported in ActiveRecord by default
* Adjust Postgres 11+ feature examples to reflect more typical real world use cases

### CODE REVIEW

**Comment by machty** [comment]
Seeing a lot of extra commits in this PR, but the README change looks good and clear to me.

**Comment by rkrage** [comment]
Yeah the diff of this PR looks fine, so I can just squash and merge. I _do_ think we should do some testing with the composite_primary_keys gem. I'd like to make sure that it actually works with pg_party before we make that recommendation. @machty can you try adding the gem to your project?

**Comment by machty** [comment]
@rkrage I got something more or less working on Rails 5.2 - it took me a while to make sense of the interplay between the gems but I was surprised in the end that it seemed to work with only a few breaking specs.

On Rails 6 I ran into some issues presumably due to the CPK gem where a newly-saved model wasn't reporting its (composite) ID correctly.

Either way, yes, I think testing and instructions to integrate with CPK would be very welcome, but I personally can't put too much more time into exploring this at the moment, already exhausted my exploratory time budget.

**Comment by webandtech** [comment]
I could go either way with even referencing the cpk gem... to me pg_party provides the helpers to leverage all the Postgres partitioning features however you want to...but dealing with specific schema decisions like what someone chose as their pks is probably ok to leave out rather than take on more dependencies...

**Comment by rkrage** [comment]
Thanks @webandtech. Okay, then for now, maybe we should just mention that CPK is a potential solution but not fully tested with this gem. I can dig into the Rails 6 issue on Friday and we can refine the recommendation from there

**Comment by rkrage** [comment]
I finally have a chance to look into this. And sorry I actually meant to thank @machty as well for doing that exploration 😃 

I'll try to get this, or a variation of this, merged today

**Review by trev — APPROVED** [review]
Nice one. The readme initially confused me as well.

File: README.md [hunk 1]
```diff
@@ -232,7 +236,11 @@ class CreateSomeHashRecord < ActiveRecord::Migration[5.1]
     # symbol is used for partition keys referring to individual columns
     # create_with_primary_key: true, template: false on Postgres 11 will rely on PostgreSQL's native partition schema
     # management vs this gem's template tables
-    create_hash_partition :some_hash_records, partition_key: :id, create_with_primary_key: true, template: false do |t|
+    # Note composite primary keys will require a workaround in ActiveRecord, such as through the use of the composite_primary_keys gem
+    create_hash_partition :some_hash_records, partition_key: :account_id, primary_key: [:id, :account_id],
+    create_with_primary_key: true, template: false do |t|
+      t.bigint :id, null: false, auto_increment: true
```
Comment (cupatea) [inline README.md:242]: `syntax error at or near "AUTO_INCREMENT"`
```suggestion
      t.bigint :id, null: false, serial: true
```

**Comment by rkrage** [comment]
I just really haven't had enough time to fully test the composite primary key gem across all versions of Rails ☹️ 

But I think this is worded in a way that suggests that it's a _potential_ solution for composite primary keys. Gonna go ahead and merge this

**Pull request merged**